### PR TITLE
[MM-69457] Stop logging PII in exportable app logs

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary

App logs are captured to rolling files that users can export via **Report a Problem** (TurboLogger `captureConsole`), and the same args are forwarded as **Sentry breadcrumbs**. Several call sites logged PII into both destinations.

This PR covers the two scoped workstreams agreed with the team:

**Fix known high-risk call sites**
- `app/init/push_notifications.ts` — no longer logs the push token value (it's a credential)
- `app/init/calls_native.ts` — logs `uuid` + presence booleans instead of the full incoming-call payload (drops `callerName`)
- `app/actions/local/channel.ts` — logs channel id + `type` instead of `displayName` (DM/GM names embed usernames)
- `app/actions/remote/channel.ts` — logs channel id + `type` instead of `display_name`
- `app/client/websocket/index.ts` — logs `code`/`reason` instead of the raw url + event object
- Swept stray raw error/event logging through `getFullErrorMessage()` and removed a `JSON.stringify(event)` dump in `app/queries/servers/system.ts`, `app/actions/websocket/files.ts`, `app/components/markdown/error_boundary.tsx`

**Prevent reintroduction**
- `eslint.config.mjs` — error-level `no-restricted-syntax` rule flagging PII (names/emails/usernames) and credentials (tokens/passwords/secrets) passed to `logError/logWarning/logInfo/logDebug`
- `CLAUDE.md` — added "Logging & PII" guidance (logs are user-exportable + sent to Sentry; log IDs not names/tokens; wrap errors with `getFullErrorMessage()`)

Structural follow-ups (narrowing `captureConsole`, a write-time redaction backstop, export-time consent UX) are intentionally left to a separate issue. `serverUrl` was deliberately excluded from the lint rule — it's logged widely as diagnostic context and including it would break many existing call sites; out of scope here.

#### Ticket Link

[MM-69457](https://mattermost.atlassian.net/browse/MM-69457)

#### Device Information

No UI changes

#### Release Note

```release-note
Fixed an issue where exported mobile logs could include personally identifiable information such as user display names and push notification tokens.
```

🤖 Generated with [Claude Code](https://claude.com/claude-code)

[MM-69457]: https://mattermost.atlassian.net/browse/MM-69457?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/actions/local/channel.ts [hunk 1]
```diff
@@ -100,7 +100,7 @@ export async function switchToChannel(serverUrl: string, channelId: string, team
                     await dismissAllRoutesAndPopToScreen(Screens.CHANNEL);
                 }
 
-                logInfo('channel switch to', channel?.displayName, channelId, (Date.now() - dt), 'ms');
+                logInfo('channel switch to', channelId, 'type', channel?.type, (Date.now() - dt), 'ms');
```
Comment (lieut-data) [inline app/actions/local/channel.ts:103]: I'm not sure of mobile logging conventions, but parsing `channel switch to dkrhkzcjd3gn8xc8h4nyaaf4uo public 200ms' feels slightly harder than below on line 106 where we appear to name the fields explicitly.

Should we try to converge on something like this?
```suggestion
                logInfo('channel switch to channel id: ', channelId, 'type: ', channel?.type, 'time: ', (Date.now() - dt), 'ms');
```

File: app/client/websocket/index.test.ts [hunk 2]
```diff
@@ -187,9 +187,11 @@ describe('WebSocketClient', () => {
 
         await client.initialize();
 
-        mockConn.onError.mock.calls[0][0]({url: 'wss://example.com/api/v4/websocket'});
+        mockConn.onError.mock.calls[0][0]({url: 'wss://example.com/api/v4/websocket', code: 1006, reason: 'abnormal closure'});
 
-        expect(logError).toHaveBeenCalledWith('websocket error', 'wss://example.com/api/v4/websocket');
+        // The URL is not logged (it is sensitive); only the code/reason are.
```
Comment (lieut-data) [inline app/client/websocket/index.test.ts]: Should we strip the domain instead of suppressing the log altogether in case a misconfiguration or proxy server interference?

File: app/init/calls_native.ts [hunk 3]
```diff
@@ -85,7 +85,7 @@ class CallsNativeSingleton {
         const {uuid, serverId, channelId, postId, threadId} = payload;
 
         if (!serverId || !channelId) {
-            logDebug('onIncomingCall received without serverId/channelId; ignoring', payload);
+            logDebug('onIncomingCall received without serverId/channelId; ignoring', uuid, {hasServerId: Boolean(serverId), hasChannelId: Boolean(channelId)});
             return;
         }
```
Comment (lieut-data) [inline app/init/calls_native.ts:94]: Could we just split these up into two `if` blocks and simplify?

**Review by lieut-data — APPROVED** [review]
A few thoughts, but not strictly blocking.

File: app/actions/local/channel.ts (hunk 1 above)
Comment (enahum) [reply app/actions/local/channel.ts:103]: There are't really any conventions, both are good, this log is mainly to detect if there is a perf downgrade in channel switching.

That said, perhaps we should come up with some conventions across the three tiers and apply that in a future PR

File: app/actions/websocket/files.ts [hunk 4]
```diff
@@ -3,17 +3,18 @@
...
 
 export async function handleFileDownloadRejected(serverUrl: string, msg: WebSocketMessage) {
     try {
         const data = msg.data as FileDownloadRejectedEvent;
 
-        logDebug('[handleFileDownloadRejected] Received event data:', JSON.stringify(data));
-
         const {file_id: fileId, rejection_reason: rejectionReason, download_type: downloadType} = data;
 
+        logDebug('[handleFileDownloadRejected] Received event for download type', downloadType, 'hasFileId', Boolean(fileId));
```
Comment (enahum) [inline app/actions/websocket/files.ts]: Perhaps good to add the reason as well

File: app/client/websocket/index.ts [hunk 5]
```diff
@@ -298,8 +298,7 @@ export default class WebSocketClient {
         this.conn!.onError((evt: any) => {
             if (evt.url === this.url) {
                 if (this.connectFailCount <= 1) {
-                    logError('websocket error', this.url);
-                    logError('WEBSOCKET ERROR EVENT', evt);
+                    logError('websocket error', 'code', evt?.code, 'reason', evt?.reason);
```
Comment (enahum) [inline app/client/websocket/index.ts]: Ahh you see.. here is important to know the server

File: app/init/push_notifications.ts [hunk 6]
```diff
@@ -323,7 +323,7 @@ class PushNotificationsSingleton {
 
             const token = `${prefix}-v2:${deviceToken}`;
             storeDeviceToken(token);
-            logDebug('Notification token registered', token);
+            logDebug('Notification token registered');
```
Comment (enahum) [inline app/init/push_notifications.ts]: Redact the token don't strip it entirely.. keep the prefix redact deviceId to first 16 chars

**Review by enahum — CHANGES_REQUESTED** [review]

File: app/actions/websocket/files.ts (hunk 4 above)
Comment (hanzei) [reply app/actions/websocket/files.ts]: `rejectionReason`  is a plugin-supplied string that can contain arbitrary content. I'm hesitant to add that to the logs, as the customer might not want to share it.

File: app/client/websocket/index.ts (hunk 5 above)
Comment (hanzei) [reply app/client/websocket/index.ts]: Good call. I think we are good to log the server URL. It's already contained in the support packet as well.

File: app/client/websocket/index.test.ts (hunk 2 above)
Comment (hanzei) [reply app/client/websocket/index.test.ts]: I ended up adding the domain back again. The server URL is also included in the Support Packet. I don't think there are major concerns with logging it.

File: app/init/calls_native.ts (hunk 3 above)
Comment (hanzei) [reply app/init/calls_native.ts:94]: Good call, fixed that

File: app/init/push_notifications.ts (hunk 6 above)
Comment (hanzei) [reply app/init/push_notifications.ts]: Logging a prefix is fine. Added that.

File: app/actions/local/channel.ts (hunk 1 above)
Comment (hanzei) [reply app/actions/local/channel.ts:103]: I would love to see convention for the log entries. But it's out of scope for this PR.

Though, I did apply the suggestion from @lieut-data for readability.

**Review by edgarbellot — APPROVED** [review]
Great work! 🚀  Did some tests and the main paths + edge cases are covered :)

**Comment by hanzei** [comment]
@enahum Gentle ping to review the PR :wink:

**Review by enahum — APPROVED** [review]
